src/posix/platform/udp.cpp

```diff
@@ -408,6 +422,8 @@ void platformUdpInit(const char *aIfName)
     {
         perror("if_nametoindex");
     }
+
+    sPlatNetifName = aIfName;
```

Comment:
Platform netif name is used or going to be used by a number of platform modules (e.g. udp, uart).
But saving the pointer to the Netif name isn't the safest way. The pointer `aIfName` does point to static memory in the current implementation, but not guaranteed so. 
I added a posix API `platformGetNetifName()` in #5376 so that `udp` and `uart` can use this interface to get the Netif name. Do you think it's a better way? @bukepo